game/world/waymatrix.cpp

```diff
@@ -178,11 +181,51 @@ void WayMatrix::marchPoints(DbgPainter &p) const {
 
 void WayMatrix::adjustWaypoints(std::vector<WayPoint> &wp) {
   for(auto& w:wp) {
-    w.y = world.physic()->landRay(w.position()).v.y;
+    auto ray = world.physic()->landRay(w.position());
+    if(ray.hasCol)
+      w.y = ray.v.y;
     indexPoints.push_back(&w);
     }
   }
 
+void WayMatrix::calculateLadderPoints() {
+  float dist   = 100.f;
+  Vec3  offset = {3.f,0,3.f};
```

Comment:
Hm, padding for 3 centimeters? Is it really that impactful? 

also `static const`

Reply:
There's one double ladder which fails otherwise. But i don't think any npc uses that ladder and it's not working in vanilla. So could remove it.